### PR TITLE
Rehash of minor changes from #1137

### PR DESCRIPTION
**Pull request by asmodai27**
# Description
In an effort to rework the architecture of functionalities from #1137 (and following WIP PRs in the chain), I started with splitting those minor changes (mostly unrelated to uart-dma or framing) from the main ones, which will hopefully make the next PR (with functional changes) easier to follow.

# General checklist:

- [x] I have performed a self-review, including - looked through the `Files changed` tab, browsed repository in my branch
- [x] I have made corresponding changes to the documentation - to reflect changes in code, electrical or mechanical design
- [x] All new documentation or graphics follow the [documentation style guide](https://github.com/RespiraWorks/Ventilator/wiki/Documentation-style-guide)
- [x] All new content is linked, easily discoverable, does not require too many clicks
- [x] I have reviewed any other relevant parts of our [contributor wiki](https://github.com/RespiraWorks/Ventilator/wiki) to ensure that this PR conforms to the standards
- [x] I have given the PR a descriptive name (prefixed with **WIP** if it is not yet ready for review)
- [x] I have tagged relevant reviewers

## Code checklist:

- [x] All new code follows our [code style](https://github.com/RespiraWorks/Ventilator/wiki/Code-style)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have described tests I used to verify my changes, instructions for reproducing them and any relevant configuration details.
- [x] I ran the unit test suite and verified that all tests pass - new and old, locally and on CI
- [x] I performed a clean build and verified that there were no warnings
- [x] I ran our static analysis tools and verified there were no warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All source files have license headers

### CODE REVIEW

File: software/controller/README.md [hunk 1]
```diff
@@ -33,32 +29,29 @@ The part of the code specific to the controller resides here.
...
-Platformio has a CLI and an IDE. You'll need the platformio CLI in order to build
-from the command-line, even if you also install the IDE.
+We use [platformio](https://platformio.org/) for building the controller code. Platformio has a CLI and an IDE. You'll need the platformio CLI in order to build from the command-line, even if you also install the IDE.
 
 Instructions for installing:
  * [CLI](https://docs.platformio.org/en/latest/core/index.html)
  * [IDE](https://docs.platformio.org/en/latest/integration/ide/pioide.html)
 
-Note that an issue prevents platformio CLI 5.0 and 5.0.1 to run our unit tests. If you are using one of these versions, you will need to upgrade to 5.0.2 or newer version using:
+Some issues may prevent specific version sof platformio from building or running unit tests. It is currently recommended that you use `v5.2.5` (latest version at the time of this writing), which you may install with i.e.:
```
Comment (martukas) [inline software/controller/README.md:44]: "version sof"

**Review by martukas — APPROVED** [review]
Just a couple of small things. Otherwise LGTM

File: software/controller/README.md [hunk 2]
```diff
@@ -6,15 +6,11 @@ Please also see the [Software design pages](../design/controller_architecture.md
...
-no harm is done to the patient.
+The ventilator actuators must be driven by control loops ensuring continuous breathing according to doctor-provided parameters, while also ensuring no harm is done to the patient.
 
-Controller code is separated from the user interface code. This makes the code base
-smaller, easier to maintain, and lessens the risk of unexpected behavior.
+Controller code is separated from the user interface code. This makes the code base smaller, easier to maintain, and lessens the risk of unexpected behavior.
 
-The controller shares the [common communications code](../common) with the GUI.
-The part of the code specific to the controller resides here.
+The controller shares the [common communications code](../common) with the GUI. The part of the code specific to the controller resides here.
```
Comment (asmodai27) [inline software/controller/README.md:13]:
```suggestion
The controller shares the [common code](../common) with the GUI. The part of the code specific to the controller resides here.
```

File: software/controller/README.md (hunk 1 above)
Comment (asmodai27) [inline software/controller/README.md:44]:
```suggestion
Some issues may prevent specific versions of platformio from building or running unit tests. It is currently recommended that you use `v5.2.5` (latest version at the time of this writing), which you may install with i.e.:
```

**Review by asmodai27 — COMMENTED** [review]
Made those into suggestions for ease of merging

**Pull request merged**